Windows: building samples reports "UnicodeDecodeError: 'gbk' codec can't decode byte 0xa2"

<!--
**Notes**
Github Discussions (https://github.com/zephyrproject-rtos/zephyr/discussions)
are available to first verify that the issue is a genuine Zephyr bug and not a
consequence of Zephyr services misuse.

This issue list is only for bugs in the main Zephyr code base
(https://github.com/zephyrproject-rtos/zephyr/). If the bug is for a project
fork (such as NCS) specific feature, please open an issue in the fork project
instead.
-->

**Describe the bug**
<!--
A clear and concise description of what the bug is.

Please also mention any information which could help others to understand
the problem you're facing:
 - What target platform are you using?
 - What have you tried to diagnose or workaround this issue?
 - Is this a regression? If yes, have you been able to "git bisect" it to a
   specific commit?
 - ...
-->
After doing the "west update" and executing the "pip3 install -r %HOMEPATH%\zephyrproject\zephyr\scripts\requirements.txt", we cannot build the samples. It always report "UnicodeDecodeError: 'gbk' codec can't decode byte 0xa2 in position 46".

**To Reproduce**
<!--
Steps to reproduce the behavior:
1. mkdir build; cd build
2. cmake -DBOARD=board\_xyz
3. make
4. See error
-->
1. PS C:\ambiqcodes\ambiqzephyr\zephyr> west update
2. PS C:\ambiqcodes\ambiqzephyr\zephyr> pip3 install -r .\scripts\requirements.txt
3. PS C:\ambiqcodes\ambiqzephyr\zephyr> west build -b apollo4p_evb .\samples\hello_world\ -p

**Expected behavior**
<!--
A clear and concise description of what you expected to happen.
-->
No compiling error reports.

**Impact**
<!--
What impact does this issue have on your progress (e.g., annoyance, showstopper)
-->
We cannot continue to develop until this issue getting resolved.

**Logs and console output**
<!--
If applicable, add console logs or other types of debug information
e.g Wireshark capture or Logic analyzer capture (upload in zip archive).
copy-and-paste text and put a code fence (\`\`\`) before and after, to help
explain the issue. (if unable to obtain text log, add a screenshot)
-->
PS C:\ambiqcodes\ambiqzephyr> west build -b apollo4p_evb .\zephyr\samples\hello_world\ -p
-- west build: making build dir C:\ambiqcodes\ambiqzephyr\build pristine
-- west build: generating a build system
Loading Zephyr default modules (Zephyr base).
-- Application: C:/ambiqcodes/ambiqzephyr/zephyr/samples/hello_world
-- CMake version: 3.22.3
-- Found Python3: C:/Python310/python.exe (found suitable version "3.10.5", minimum required is "3.10") found components: Interpreter
-- Cache files will be written to: C:/ambiqcodes/ambiqzephyr/zephyr/.cache
-- Zephyr version: 3.7.99 (C:/ambiqcodes/ambiqzephyr/zephyr)
-- Found west (found suitable version "1.0.0", minimum required is "0.14.0")
CMake Error at C:/ambiqcodes/ambiqzephyr/zephyr/cmake/modules/boards.cmake:198 (message):
  Error finding board: apollo4p_evb
 
  Error message: Traceback (most recent call last):
 
    File "C:\ambiqcodes\ambiqzephyr\zephyr\scripts\list_boards.py", line 387, in <module>
      dump_v2_boards(args)
    File "C:\ambiqcodes\ambiqzephyr\zephyr\scripts\list_boards.py", line 335, in dump_v2_boards
      boards = find_v2_boards(args)
    File "C:\ambiqcodes\ambiqzephyr\zephyr\scripts\list_boards.py", line 262, in find_v2_boards
      b = load_v2_boards(args.board, board_yml, systems)
    File "C:\ambiqcodes\ambiqzephyr\zephyr\scripts\list_boards.py", line 187, in load_v2_boards
      b = yaml.load(f.read(), Loader=SafeLoader)
 
  UnicodeDecodeError: 'gbk' codec can't decode byte 0xa2 in position 46:
  illegal multibyte sequence

**Environment (please complete the following information):**

 - OS: Windows 10
 - Toolchain: Zephyr SDK 0.16.8
 - Commit SHA or Version used: 1ec5ce05f9f84b18cb8056d01917480a6ec39e52 (latest main branch)

**Additional context**
<!--
Add any other context that could be relevant to your issue, such as pin setting,
target configuration, ...
-->

--- a/scripts/ci/test_plan.py
+++ b/scripts/ci/test_plan.py
@@ -246,7 +246,7 @@ class Filters:
                 c = (zephyr_base / changed).resolve()
                 if c.is_relative_to(board.dir.resolve()):
                     for file in glob.glob(os.path.join(board.dir, f"{board.name}*.yaml")):
-                        with open(file, 'r') as f:
+                        with open(file, 'r', encoding='utf-8') as f:
                             b = yaml.load(f.read(), Loader=SafeLoader)
                             matched_boards[b['identifier']] = board
 

--- a/scripts/list_boards.py
+++ b/scripts/list_boards.py
@@ -183,7 +183,7 @@ def find_arch2board_set_in(root, arches, board_dir):
 def load_v2_boards(board_name, board_yml, systems):
     boards = []
     if board_yml.is_file():
-        with board_yml.open('r') as f:
+        with board_yml.open('r', encoding='utf-8') as f:
             b = yaml.load(f.read(), Loader=SafeLoader)
 
         try:

--- a/scripts/list_hardware.py
+++ b/scripts/list_hardware.py
@@ -188,7 +188,7 @@ def find_v2_archs(args):
         archs_yml = root / ARCHS_YML_PATH
 
         if Path(archs_yml).is_file():
-            with Path(archs_yml).open('r') as f:
+            with Path(archs_yml).open('r', encoding='utf-8') as f:
                 archs = yaml.load(f.read(), Loader=SafeLoader)
 
             try:

--- a/scripts/utils/board_v1_to_v2.py
+++ b/scripts/utils/board_v1_to_v2.py
@@ -74,7 +74,7 @@ def board_v1_to_v2(board_root, board, new_board, group, vendor, soc, variants):
             }
         }
     else:
-        with open(board_settings_file) as f:
+        with open(board_settings_file, encoding='utf-8') as f:
             yaml = ruamel.yaml.YAML(typ='safe', pure=True)
             board_settings = yaml.load(f) # pylint: disable=assignment-from-no-return
 

--- a/scripts/zephyr_module.py
+++ b/scripts/zephyr_module.py
@@ -193,7 +193,7 @@ def process_module(module):
     for module_yml in [module_path / MODULE_YML_PATH,
                        module_path / MODULE_YML_PATH.with_suffix('.yaml')]:
         if Path(module_yml).is_file():
-            with Path(module_yml).open('r') as f:
+            with Path(module_yml).open('r', encoding='utf-8') as f:
                 meta = yaml.load(f.read(), Loader=SafeLoader)
 
             try:

